### PR TITLE
buf: stream and iter helpers

### PR DESCRIPTION
**Pull request by carllerche**
Add utilities for creating `BufStream` values from stream and iterator values.

### CODE REVIEW

File: tokio-buf/src/util/iter.rs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    _p: PhantomData<E>,
+}
+
+impl<I, E> BufStream for Iter<I, E>
+where
+    I: Iterator,
+    I::Item: Buf,
+{
+    type Item = I::Item;
+    type Error = E;
```
Comment (LucioFranco) [inline tokio-buf/src/util/iter.rs]: not totally sure but does it make sense to have an error here or just use a Never error type?

Comment (carllerche) [reply tokio-buf/src/util/iter.rs]: Good question... having an error would mean it doesn't have to be mapped... but you are probably right.

**Comment by carllerche** [comment]
@LucioFranco I added some quick docs. Any thoughts?

**Comment by LucioFranco** [comment]
@carllerche read through them they are great! 👍

**Pull request merged**